server/sonar-web/src/main/js/api/branches.ts

```diff
@@ -17,9 +17,17 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-import { getJSON } from '../helpers/request';
+import { getJSON, post } from '../helpers/request';
 import throwGlobalError from '../app/utils/throwGlobalError';
 
 export function getBranches(project: string): Promise<any> {
   return getJSON('/api/project_branches/list', { project }).then(r => r.branches, throwGlobalError);
 }
+
+export function deleteBranch(project: string, branch: string): Promise<void | Response> {
+  return post('/api/project_branches/delete', { project, branch }).catch(throwGlobalError);
+}
+
+export function renameBranch(project: string, branch: string): Promise<void | Response> {
+  return post('/api/project_branches/rename', { project, branch }).catch(throwGlobalError);
```

Comment:
The same is happening when trying to rename and switching the browser offline... No error displayed and infinite spinner.